### PR TITLE
Update release tool with new merge strategy

### PR DESCRIPTION
**Pull request by droidmonkey**
* Fast-forward `latest` tag to merged tag
* Don't merge into TARGET_BRANCH
* Fix missing argon2 in snapcraft.yml

[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )

NOTE: I had to use `grep -Pzo` on MSYS2 and Linux to pull the CHANGELOG text for the tag. On Mac it uses `grep -Ezo` since -P is not available.

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tested on MSYS2

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change that fixes an issue)

### CODE REVIEW

File: release-tool [hunk 1]
```diff
@@ -540,24 +515,28 @@ merge() {
         fi
     fi
 
-    CHANGELOG=$(grep -Ezo "## ${RELEASE_NAME} \([0-9]{4}-[0-9]{2}-[0-9]{2}\)\n\n(.|\n)+?\n\n## " CHANGELOG.md \
+    local flags="-Pzo"
+    if OS_MACOS; then
+        flags="-Ezo"
+    fi
+    CHANGELOG=$(grep ${flags} "## ${RELEASE_NAME} \([0-9]{4}-[0-9]{2}-[0-9]{2}\)\n\n(.|\n)+?\n\n## " CHANGELOG.md \
```
Comment (droidmonkey) [inline release-tool]: I had to use -P on MSYS2 to make this work at all. On macOS it uses -E.

File: release-tool [hunk 2]
```diff
@@ -279,7 +277,7 @@ checkGitRepository() {
 }
 
 checkReleaseDoesNotExist() {
-    if ! git tag | grep -q "^$TAG_NAME$"; then
+    if [ $(git tag -l $TAG_NAME) ]; then
```
Comment (phoerious) [inline release-tool:280]: I think this quote-less syntax caused errors on macOS last time.
Maybe do
```suggestion
    if git rev-parse "$TAG_NAME" &>/dev/null; then
```

File: release-tool (hunk 1 above)
Comment (phoerious) [reply release-tool]: The two syntaxes are not fully compatible, but I can see that GNU grep seems to be ignoring the `-z` somehow. Not sure what's going on there.

File: release-tool [hunk 3]
```diff
@@ -540,24 +515,28 @@ merge() {
         fi
     fi
 
-    CHANGELOG=$(grep -Ezo "## ${RELEASE_NAME} \([0-9]{4}-[0-9]{2}-[0-9]{2}\)\n\n(.|\n)+?\n\n## " CHANGELOG.md \
+    local flags="-Pzo"
+    if OS_MACOS; then
```
Comment (phoerious) [inline release-tool]: This is definitely an error. Use
```suggestion
    if [ -n "$OS_MACOS" ]; then
```
instead.

**Review by phoerious — CHANGES_REQUESTED** [review]

File: release-tool (hunk 2 above)
Comment (droidmonkey) [reply release-tool:280]: I just tested this on macOS and it worked fine, I much prefer the direct git tag method than the rev-parse craziness.
`if [ $(git tag -l 2.7.0-beta1) ]; then echo 1; fi`

**Review by phoerious — APPROVED** [review]

**Pull request merged**